projects/plugins/jetpack/class.json-api.php

```diff
@@ -202,12 +202,21 @@ public function add( WPCOM_JSON_API_Endpoint $endpoint ) {
 	}
 
 	/**
-	 * Determine if a string is truthy.
+	 * Determine if a string is truthy. If it's not a string, which can happen with
+	 * not well-formed data coming from Jetpack sites, we still consider it a truthy value.
 	 *
-	 * @param string $value "1", "t", and "true" (case insensitive) are falsey, everything else isn't.
+	 * @param string $value "1", "t", and "true" (case insensitive) are truthy, everything else isn't.
 	 * @return bool
 	 */
 	public static function is_truthy( $value ) {
+		if ( true === $value ) {
+			return true;
+		}
+
+		if ( ! is_string( $value ) ) {
```

Comment:
@zinigor, should this perhaps be `if ( ! is_string( $value ) ) || is_integer( $value ) )`?

In calypso, we a have screen, where toggling off the check button in the screenshot, sends a payload like `shared : 1` to the end-point. The `is_truthy` function returns false when the value is 1.  

I have tweaked calypso to send true or false to fix our issue, but maybe we should handle this differently for integers? 

<img width="1152" alt="Screenshot 2024-01-23 at 5 15 45 PM" src="https://github.com/Automattic/jetpack/assets/6594561/6140369f-b918-4255-bc3a-28a7eccd00e2">